lib/Qpsmtpd/Plugin.pm

```diff
@@ -347,10 +347,16 @@ sub _register_standard_hooks {
     }
 }
 
+sub db_args {
+    my ( $self, %arg ) = @_;
```

Comment:
this seems likely to encounter problems when the wrong number of arguments is passed in (due to an error elsewhere). Validating the arguments (at least the correct qty) seems prudent.